tools/dashboard/generate_dashboard.py

```diff
@@ -19,71 +21,157 @@
+
+
+# ======================================================================================
+class GitSha(str):
+    pass  # use only for type annotations
+
+
+# ======================================================================================
+class Commit:
+    def __init__(
+        self,
+        sha: GitSha,
+        author_name: str,
+        author_email: str,
+        date: datetime,
+        message: str,
+    ):
+        self.sha = sha
+        self.author_name = author_name
+        self.author_email = author_email
+        self.date = date
+        self.message = message
+
+
+# ======================================================================================
+class PlotCurve:
+    def __init__(self, label: str):
+        self.label = label
+        self.x: List[float] = []
+        self.y: List[float] = []
+        self.yerr: List[float] = []
+
+
+# ======================================================================================
+class Plot:
+    def __init__(self, name: str, title: str, ylabel: str, **_: Any):
+        self.name = name
+        self.title = title
+        self.ylabel = ylabel
+        self.curves_by_name: Dict[str, PlotCurve] = {}
+
+    @property
+    def curves(self) -> ValuesView[PlotCurve]:
+        return self.curves_by_name.values()
+
+
+# ======================================================================================
+class PlotPoint:
+    def __init__(
+        self, plot: str, name: str, label: str, y: float, yerr: float, **_: Any
+    ):
+        self.plot_name = plot
+        self.curve_name = name
+        self.curve_label = label
+        self.y = y
+        self.yerr = yerr
+
+
+# ======================================================================================
+class Report:
+    Status = Literal["OK", "FAILED", "OUTDATED", "UNKNOWN"]
+
+    def __init__(
+        self,
+        status: Status,
+        summary: str,
+        sha: Optional[GitSha] = None,
+        plots: Optional[List[Plot]] = None,
+        plot_points: Optional[List[PlotPoint]] = None,
+    ):
+        self.status = status
+        self.sha = sha
+        self.summary = summary
+        self.plots = plots if plots else []
+        self.plot_points = plot_points if plot_points else []
+        self.archive_path: Optional[str] = None
+
+
+# ======================================================================================
+class Status:
+    def __init__(self, last_ok: Optional[GitSha], notified: bool):
+        self.last_ok = last_ok
+        self.notified = notified
 
 
 # ======================================================================================
-def main():
+def main() -> None:
     if len(sys.argv) < 4:
         usage = "<config-file> <status-file> <output-dir> [--send-emails]"
         print(f"Usage: generate_dashboard.py {usage}")
         sys.exit(1)
 
     config_fn, status_fn, outdir = sys.argv[1:4]
-    outdir = outdir.strip("/")
+    outdir = outdir.rstrip("/")
```

Comment:
using the builtin `argparse` instead of handrolling and then using `type=pathlib.Path` might avoid some headaches when it comes to raw path manipulation, see also https://docs.python.org/3/library/argparse.html#type